File: pyforms/web/Controls/ControlSlider.py
from pyforms.web.Controls.ControlBase import ControlBase


class ControlSlider(ControlBase):

	_min = 0
	_max = 100

	def __init__(self, label = "", defaultValue = 0, min = 0, max = 100):
		self._updateSlider = True
		self._min = min
		self._max = max
		
		ControlBase.__init__(self, label, defaultValue)
		
	def initControl(self): return "new ControlSlider('{0}', {1})".format( self._name, str(self.serialize()) )

	def valueChanged(self, value):
		self._updateSlider = False
		self.value = value
		self._updateSlider = True


	def changed(self): pass
	

	@property
	def min(self): return self._min

	@min.setter
	def min(self, value):  self._min = value

	@property
	def max(self):  return  self._max

	@max.setter
	def max(self, value):  self._max = value


	def serialize(self):
		data = ControlBase.serialize(self)
		data.update({ 'max': self.max, 'min': self.min })
		return data
		
	def deserialize(self, properties):
		ControlBase.deserialize(self,properties)
		self.max = properties[u'max']
		self.min = properties[u'min']
		

File: pyforms/web/django/static/pyformsjs/ControlSlider.js


function ControlSlider(name, properties){
	ControlBase.call(this, name, properties);
};
ControlSlider .prototype = Object.create(ControlBase.prototype);

////////////////////////////////////////////////////////////////////////////////

ControlSlider.prototype.get_value = function(){ 
	return this.jquery().val();
};

////////////////////////////////////////////////////////////////////////////////

ControlSlider.prototype.set_value = function(value){
	this.jquery().val(value);
	$( "#value"+this.name ).html( value );
};

////////////////////////////////////////////////////////////////////////////////

ControlSlider.prototype.init_control = function(){
	var html = 	"<div id='"+this.place_id()+"' class='field ControlSlider' title='"+this.properties.help+"' >";
	html += 	"<label>"+this.properties.label;
	html += 	" <div id='value"+this.control_id()+"' class='ui basic label'>"+this.properties.value+"</div>";
	html += 	"</label>";
	html += 	"<input style='width:100%;' type='range' name='"+this.name+"' value='"+this.properties.value+"' id='"+this.control_id()+"' min='"+this.properties.min+"' max='"+this.properties.max+"'>";
	html += 	"</div>";

	this.jquery_place().replaceWith(html);
	//this.jquery().on('input', function () {$(this).trigger('change');});
	var self = this;
	this.jquery().change(function(){ 
		$( "#value"+self.control_id() ).html( $(this).val() ); 
		self.basewidget.fire_event( self.name, 'changed' );
	});

	if(this.properties.visible) 
		this.jquery_place().show();
	else 
		this.jquery_place().hide();
};

////////////////////////////////////////////////////////////////////////////////

ControlSlider.prototype.deserialize = function(data){
	this.properties = $.extend(this.properties, data);
	this.set_value(this.properties.value);
};

////////////////////////////////////////////////////////////////////////////////

ControlSlider.prototype.serialize = function(){
	this.properties.value = this.get_value();
	return this.properties; 
};

////////////////////////////////////////////////////////////////////////////////